adapters/src/test/java/com/squareup/moshi/adapters/RuntimeJsonAdapterFactoryTest.java

```diff
@@ -181,6 +183,27 @@
     }
   }
 
+  @Test public void usesObjectAdapterFromJson() throws IOException {
+    Moshi moshi = new Moshi.Builder()
+        .add(RuntimeJsonAdapterFactory.of(Message.class, "type")
+            .registerSubtype(Success.class, "success"))
+        .add(new Object() {
+          @FromJson Object fromJson(JsonReader reader, JsonAdapter<Object> delegate)
```

Comment:
>maybe just hardcode an object adapter to always call skipValue() and then return "foo" and confirm that’s what happens?

we can't quite do it that simply because we still need the object adapter to handle maps and strings correctly, at least, so we can get the "type" out of the JSON object.

not sure how to simplify this further. i could remove the `new BigDecimal(reader.nextString())` and return 0 or something in this example, if that's helpful.